tests/io/v3/was/scans/test_scans_api.py

```diff
@@ -144,5 +331,133 @@ def test_update_status(api):
+            'limit': 200,
+            'sort': [{
+                'name': 'created_at',
+                'order': 'desc'
+            }]
+        },
+        'items': [{
+            'vuln_id': '3072f8d9-a0c2-443f-bca8-b47e63ebaa80',
+            'scan_id': '7f3428c5-0f5a-4812-a728-fffbcbf7c132',
+            'plugin_id': 98009,
+            'created_at': '2020-02-05T23:25:31Z',
+            'uri': 'http://192.0.2.119',
+            'is_page': False,
+            'details': {
+                'input_name': None,
+                'input_type': None,
+                'output': 'The scan has discovered 23 distinct URLs',
+                'proof': None,
+                'payload': None,
+                'selector': None,
+                'selector_url': None,
+                'signature': None,
+                'request': None,
+                'response': None
+            },
+            'attachments': [{
+                'attachment_id': 'b13a9fb5-cb0d-47d8-a6c1-063fbe6f8250',
+                'created_at': '2020-02-05T23:25:33.740Z',
+                'attachment_name': 'sitemap.csv',
+                'md5': 'md5:b2e06491f801f7f5b5f229bbf6efd7e9',
+                'file_type': 'text/plain',
+                'size': 0
+            }]
+        },
+            {
+                'vuln_id': 'a1dc9d88-44de-4f5c-9258-3dbb02baa010',
+                'scan_id': '7f3428c5-0f5a-4812-a728-fffbcbf7c132',
+                'plugin_id': 98000,
+                'created_at': '2020-01-01T23:25:31Z',
+                'uri': 'http://192.0.2.119',
+                'is_page': False,
+                'details': {
+                    'input_name': None,
+                    'input_type': None,
+                    'output': '\nEngine Version 0.41.0',
+                    'proof': None,
+                    'payload': None,
+                    'selector': None,
+                    'selector_url': None,
+                    'signature': None,
+                    'request': None,
+                    'response': None
+                },
+                'attachments': []
+        }
+        ]
+    }
+    URL = 'vulnerabilities/search?limit=200&sort=name&sort=order&offset=0'
+    responses.add(
+        responses.POST,
+        f'{WAS_SCANS_BASE_URL}/scans/{SAMPLE_SCAN_ID}/{URL}',
+        match=[matchers.json_params_matcher(payload)],
+        json=api_response
+    )
+
+    iterator = api.v3.was.scans.search_vulnerabilities(SAMPLE_SCAN_ID,
+                                                       fields=fields,
+                                                       limit=200,
+                                                       sort=sort,
+                                                       filter=filters
+                                                       )
+    assert isinstance(iterator, SearchIterator)
+
+    event_list = []
+    for event in iterator:
+        event_list.append(event)
+    assert len(event_list) == api_response['pagination']['total']
```

Comment:
Refactor above code to get rid of loop.

Reply:
removed for loop here and used `len(list(iterator))` in assert statement